shared/testgrid/testgrid_test.go

```diff
@@ -20,64 +20,54 @@ import (
 	"io/ioutil"
 	"os"
 	"testing"
+	"fmt"
+	"path"
 
 	"github.com/knative/test-infra/shared/testgrid"
+	"github.com/knative/test-infra/shared/junit"
+	"github.com/knative/test-infra/shared/prow"
 )
 
 const (
 	filename = "junit_test.xml"
 	name     = "test"
 )
 
-func checkFileText(t *testing.T, expected string) {
-	d, err := ioutil.ReadFile(filename)
+func checkFileText(resultFile, expected string) error {
+	d, err := ioutil.ReadFile(resultFile)
+	//s := strings.Trim(string(d), "\n")
+	s := string(d)
 	if err != nil {
-		t.Errorf("Failed to open test file: %v", err)
+		return fmt.Errorf("Failed to open test file: %v", err)
```

Comment:
Not sure why you changed the approach here. Failing the test in this helper function simplifies the test, as now you have to check every return value just to do the same thing.

Reply:
forgot the original purpose of this, changed it back